src/java/main/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -1376,4 +1382,8 @@ public static String getPacketType(int packetType) {
             return "UNKNOWN";
         }
     }
+
+    private boolean isRunning() {
+        return self.isRunning() && zk.isRunning();
```

Comment:
zk.isRunning() will return false when ZooKeeperServerListener notifies an error. AND is required here, because whoever is using self.isRunning() earlier(before applying this patch) should stop/exit their execution by seeing an error in the server. Does this makes sense to you?